### PR TITLE
[WIP][ADD] Dynamic FSM Order Type

### PR DESCRIPTION
**Pull request by osi-scampbell**
Allows FSM Order Types to be dynamic, can only be created/deleted by FSM Managers

### CODE REVIEW

File: fieldservice/models/__init__.py [hunk 1]
```diff
@@ -22,4 +22,6 @@
     fsm_equipment,
     fsm_order,
     fsm_route,
+    fsm_order_type,
+
```
Comment (max3903) [inline fieldservice/models/__init__.py]:
```suggestion
```

File: fieldservice/views/menu.xml [hunk 2]
```diff
@@ -155,6 +155,14 @@
...
               groups="fieldservice.group_fsm_tag"/>
 
+    <menuitem id="menu_fsm_order_type"
+              name="Order Types"
+              action="action_fsm_order_type"
+              parent="menu_fsm_config_order"
+              sequence="30"
+              groups="fieldservice.group_fsm_manager"/>
+
+
```
Comment (max3903) [inline fieldservice/views/menu.xml]:
```suggestion
```

File: fieldservice/views/fsm_order_type.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <!-- Branch Tree View -->
```
Comment (max3903) [inline fieldservice/views/fsm_order_type.xml]:
```suggestion
    <!-- FSM Order Type Tree View -->
```

File: fieldservice/views/fsm_order_type.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        <field name="name">fsm.order.type.tree</field>
+        <field name="model">fsm.order.type</field>
+        <field name="arch" type="xml">
+            <tree string="Order Types">
+                <field name="name"/>
+            </tree>
+        </field>
+    </record>
+
+    <!-- Branch Form View -->
```
Comment (max3903) [inline fieldservice/views/fsm_order_type.xml]:
```suggestion
    <!-- FSM Order Type Form View -->
```

File: fieldservice/views/fsm_order_type.xml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <record id="fsm_order_type_form_view" model="ir.ui.view">
+        <field name="name">fsm.order.type.form</field>
+        <field name="model">fsm.order.type</field>
+        <field name="arch" type="xml">
+            <form string="Branch">
```
Comment (max3903) [inline fieldservice/views/fsm_order_type.xml]:
```suggestion
            <form string="Type">
```

**Review by max3903 — APPROVED** [review]

**Comment by max3903** [comment]
@osi-scampbell You will need to bump the version number and provide the migration script to migrate the data.

**Review by hparfr — APPROVED** [review]

**Pull request merged**